### DISCUSSION TITLE
Installing Unit on Ubuntu 24.04

### DISCUSSION THREAD

**Discussion by tufelkinder** [nginx/unit#1450] [Support] [answered]
It appears that official support for Ubuntu only extends to 23.04, so when I attempted to install unit on 24.04 I ran into an issue with Python version compatibility as there's no currently released binary package for unit-python3.12.

However, I downloaded the src and compiled python3.12-unit.so and placed this in the /usr/lib/unit/modules directory. However, on startup Unit reports in the log:

`incompatible module /usr/lib/unit/modules/python3.12.unit.so
`

What should I be doing differently to run Unit on 24.04? It seems like it would be helpful to release a unit-python3.12 binary for Ubuntu 24.04, but in lieu of that, I'm interested in understanding why my workaround isn't working.

**Comment by ac000** [comment, marked as answer]
> It appears that official support for Ubuntu only extends to 23.04,

Indeed, the docs need updating (I have a patch for that), however I believe we do actually have packages for 24.04 (Noble Numbat)

So if you follow the instructions for 23.04 but change 'lunar' to 'noble' that should get you going.

> However, I downloaded the src and compiled python3.12-unit.so and placed this in the /usr/lib/unit/modules directory. However, on startup Unit reports in the log:
>
> `incompatible module /usr/lib/unit/modules/python3.12.unit.so` 
>
> What should I be doing differently to run Unit on 24.04? It seems like it would be helpful to release a unit-python3.12 binary for Ubuntu 24.04, but in lieu of that, I'm interested in understanding why my workaround isn't working.

What version of Unit did you build against? and what version did you try running against? These *must* match.

**Reply by tufelkinder** [reply]
Great, thank you! Updating the deb lines to noble corrected the issue. 

I installed Unit from the Ubuntu Lunar (23.04) repo but compiled the unit-python module from the latest release from github releases (1.33.0). I thought those versions were actually the same, but perhaps not...

**Comment by ac000** [comment]
> I installed Unit from the Ubuntu Lunar (23.04) repo but compiled the unit-python module from the latest release from github releases (1.33.0). I thought those versions were actually the same, but perhaps not...

Actually the latest version for Lunar is 1.32.1 as it is no longer supported.
